airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -912,14 +912,27 @@ def get_key(self, key: str, bucket_name: str | None = None) -> S3ResourceObject:
         :param bucket_name: the name of the bucket
         :return: the key object from the bucket
         """
+
+        def __sanitize_extra_args() -> dict[str, str]:
+            """Parse extra_args and return a dict with only the args listed in ALLOWED_DOWNLOAD_ARGS."""
+            return {
+                arg_name: arg_value
+                for (arg_name, arg_value) in self.extra_args.items()
+                if arg_name in S3Transfer(self.conn).ALLOWED_DOWNLOAD_ARGS
```

Comment:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/customizations/s3.html#boto3.s3.transfer.S3Transfer.ALLOWED_DOWNLOAD_ARGS

```suggestion
                if arg_name in S3Transfer.ALLOWED_DOWNLOAD_ARGS
```

Reply:
Strange.  I had it that way but my IDE got grumpy so I changed it to the way I have it.   I'll check it out.

Reply:
Very strange...   PyCharm still doesn't like it, but it seems to work and the now-working unit test works fine with it either way.  I've made the change.